### DISCUSSION TITLE
how to insert data into separate independant file

### DISCUSSION THREAD

**Discussion by darkomenx** [simolus3/drift#2566] [Q&A]
Hello everybody,

I would like to create fixtures files for insert data test in my database.
I try this as like reading into documentation : 

```
import 'package:drift/drift.dart';
import 'package:drift/native.dart';
import '../db/drift_database.dart';

Future<int> addDocument(DocumentCompanion entry) {
  return into(document).insert(
    filename: 'frfergre'
  );
}
```
But : 
- it tell me that "into" is not defined
- if I import package drift_database.g.dart it tell me : "The imported library '../db/drift_database.g.dart' can't have a part-of directive"
- official documentation is very poor and never notice the import necessary for example code
- search official documentation website never retrieve word in documentation (search engine not working)

All examples are based on a very poor and basic situation code that run all in main.dart file but how use this library if we want clean separate architecture ?

Very difficult to understand how use this library with this simple CRUD operation...

Can anybody enlight me to create this kind of files ?

Thank you for your support

**Comment by simolus3** [comment]
> if I import package drift_database.g.dart it tell me : "The imported library '../db/drift_database.g.dart' can't have a part-of directive"

You should import the main database file, `drift_database.dart` instead. Part files in Dart can't be imported.

> it tell me that "into" is not defined

Because it's a method on the database class, not a top-level function. You can obtain an instance of your database and then call `database.into` on it, for instance. How to obtain that instance depends on the architecture of your app, you could register it in a provider or with another package like `get_it`.

```dart
import 'package:drift/drift.dart';
import '../db/drift_database.dart';

Future<int> addDocument(YourDatabaseClass db, DocumentCompanion entry) {
  return db.into(db.document).insert(entry);
}
```

> official documentation is very poor and never notice the import necessary for example code

That's actually kind of hurtful. Writing documentation isn't easy and while I'm simplyfing some cases, I hope that the referenced examples and snippets demonstrate how to use drift in more complex scenarios as well. For instance, the [example app](https://github.com/simolus3/drift/tree/develop/examples/app) uses the database in different files.

I also don't think the drift documentation should mention that part files can't be imported - that information belongs in a Dart language guide. We mention how to use [DAOs](https://drift.simonbinder.eu/docs/advanced-features/daos/) which can be used to split a drift database across multiple files without having to use `db.` everywhere. Apart from that, managing imports to drift databases is no different than for any other Dart code, so it didn't occur to me that this should be explicitly documented. And again, there are examples showing how to do this as well.
